sdk/core/azure-core/inc/azure/core/internal/tracing/service_tracing.hpp

```diff
@@ -193,28 +191,52 @@ namespace Azure { namespace Core { namespace Tracing { namespace _internal {
     {
     }
 
-    DiagnosticTracingFactory() = default;
-    DiagnosticTracingFactory(DiagnosticTracingFactory const&) = default;
+    TracingContextFactory() = default;
+    TracingContextFactory(TracingContextFactory const&) = default;
 
     /** @brief A ContextAndSpan provides an updated Context object and a new span object
      * which can be used to add events and attributes to the span.
      */
-    using ContextAndSpan = std::pair<Azure::Core::Context, ServiceSpan>;
+    struct TracingContext
+    {
+      Azure::Core::Context Context;
+      ServiceSpan Span;
+    };
```

Comment:
nit: We might add methods to these in the future, correct? Make it a class (and add `public:`).

Reply:
The pattern we are using is that if a type needs a constructor or methods, it becomes a class, otherwise it's a struct and yes, we can add fields to a struct in the future (we can't remove them or change their type, but we can add new ones). This is the same rules for ServiceModels.